src/linux/cpu.rs

```diff
@@ -548,20 +744,42 @@ pub(crate) fn get_vendor_id_and_brand() -> (String, String) {
             .unwrap_or_default()
     }
 
+    fn get_hex_value(s: &str) -> u16 {
+        s.split(':')
+            .last()
+            .map(|x| x.trim().to_owned())
+            .filter(|x| x.starts_with("0x"))
+            .map(|x| u16::from_str_radix(&x[2..], 16).unwrap())
+            .unwrap_or_default()
+    }
+
     let mut vendor_id = None;
     let mut brand = None;
+    let mut implementer = None;
+    let mut part = None;
 
     for it in s.split('\n') {
         if it.starts_with("vendor_id\t") {
             vendor_id = Some(get_value(it));
         } else if it.starts_with("model name\t") {
             brand = Some(get_value(it));
+        } else if it.starts_with("CPU implementer") {
+            implementer = Some(get_hex_value(it));
+        } else if it.starts_with("CPU part") {
+            part = Some(get_hex_value(it));
         } else {
             continue;
         }
         if brand.is_some() && vendor_id.is_some() {
             break;
         }
+        if implementer.is_some() && part.is_some() {
```

Comment:
Why making this into a different condition? Why not having something like:

```rust
if (brand.is_some() && vendor_id.is_some()) || (implementer.is_some() && part.is_some()) {
```

or a `else if`?